lib/replicator.js

```diff
@@ -1838,15 +1838,20 @@ module.exports = class Replicator {
 
     let clear = null
 
+    const reader = this.core.storage.createReadBatch()
+
+    const blocks = new Map()
     for (const b of this._blocks) {
-      if (this.core.bitfield.get(b.index) === false) continue
+      blocks.set(b.index, this.core.blocks.get(reader, b.index))
```

Comment:
Sanity check: `this.core.blocks.get(reader, b.index)` can never reject? Otherwise I think it needs a safetyCatch or noop catch-handler, since there are async operations before the promises are awaited